src/FakeItEasy/MethodBaseExtensions.cs

```diff
@@ -18,5 +20,119 @@ public static string GetGenericArgumentsCSharp(this MethodBase method)
 
             return string.Concat("<", string.Join(", ", genericArguments.Select(type => type.FullNameCSharp()).ToArray()), ">");
         }
+
+        public static bool IsPropertyGetterOrSetter(this MethodBase method)
+        {
+            return method.IsPropertyGetter() || method.IsPropertySetter();
+        }
+
+        public static bool IsPropertySetter(this MethodBase method)
+        {
+            return method.IsSpecialName && method.Name.StartsWith("set_", StringComparison.Ordinal);
+        }
+
+        public static bool IsPropertyGetter(this MethodBase method)
+        {
+            return method.IsSpecialName && method.Name.StartsWith("get_", StringComparison.Ordinal);
+        }
+
+        public static string GetDescription(this MethodBase method)
+        {
+            var builder = new StringBuilder();
+
+            builder
+                .Append(method.DeclaringType)

```

Comment:
Just noticed that we don't include "nicely formatted" generic arguments for the declaring type in `DefaultFakeObjectCallFormatter`, so the result is a mix of CLR and C# formatting. It will go away with #1047, but in the meantime, I'll fix it both in `DefaultFakeObjectCallFormatter` and here.